.github/workflows/deploy.yaml

```
@@ -9,24 +9,49 @@ on:
     branches:
       - main
 
+concurrency:
+  # For PRs, use PR number to group concurrent runs
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+  # Only apply concurrency to the same environment
+  # This ensures we don't cancel production deployments for development PRs
+  environment: ${{ github.event_name == 'push' && 'production' || 'development' }}
```

actionlint (1.7.4)

unexpected key "environment" for "concurrency" section. expected one of "cancel-in-progress", "group"

(syntax-check)